cdap-docs/reference-manual/source/http-restful-api/reporting.rst

```diff
@@ -0,0 +1,440 @@
+         "fieldName":"<name-of-the-field-to-sort-by>"
+      }
+   ]
+ }
+
+Example request Json
+
+.. highlight:: json
+
+::
+
+ {
+   "name":"Failed, Stopped, Running, Succeeded runs - Jul 23, 2018 00:00am to Jul 24, 2018 21:00pm",
+   "start":1532329200,
+   "end":1532491200,
+   "fields":[
+      "artifactName",
+      "applicationName",
+      "program",
+      "programType",
+      "namespace",
+      "status",
+      "start",
+      "end"
+   ],
+   "filters":[
+      {
+         "fieldName":"status",
+         "whitelist":[
+            "FAILED",
+            "STOPPED",
+            "RUNNING",
+            "COMPLETED",
+            "KILLED"
+         ]
+      },
+      {
+         "fieldName":"namespace",
+         "whitelist":[
+            "default"
+         ]
+      }
+   ]
+ }
+
+On executing this generate reports endpoint, backend asynchronously launches spark job to process and generate report and immediately responds with report-id JSON.
+
+.. highlight:: json
+
+::
+
+ {
+    "id": <report-id>
+ }
+
+Users can use this report-id to check on status of the report, if it is running, completed or failed using info endpoint
+and download completed reports to get information using download endpoint.
+
+**Note** : Currently sort cannot be performed on more than one field.
+
+Listing Reports
+---------------
+
+.. highlight:: console
+
+::
+
+ GET :
+ /v3/namespaces/system/apps/ReportGenerationApp/spark/ReportGenerationSpark/methods/reports
+
+.. list-table::
+ :header-rows: 1
+ :widths: 30 70
+
+ * - Query Param
+   - Description
+ * - offset
+   - offset to read reports from
+ * - limit
+   - limit on the number of reports to return
+
+Response Json with explanation for fields
+
+.. highlight:: console
+
+::
+
+ {
+   "offset":<offset>,
+   "limit":<limit>,
+   "total":<total-reports-size-returned>,
+   "reports":[
+      {
+         "id":<report-id>,
+         "name":<report-name>,
+         "created":<creation-time-seconds>,
+         "expiry":<report-expiry-time-seconds>,
+         "status":<report-status>
+      },
+      ...
+   ]
+ }
+
+Example Response JSON
+
+.. highlight:: json
+
+::
+
+ {
+   "offset":0,
+   "limit":20,
+   "total":7,
+   "reports":[
+      {
+         "id":"18a782b5-8fd4-11e8-ae71-acde48001122",
+         "name":"Failed, Succeeded, Running, Stopped runs - Jul 24, 2018 22:29pm to Jul 24, 2018 23:29pm",
+         "created":1532500174,
+         "expiry":1532672986,
+         "status":"COMPLETED"
+      },
+      ...
+      {
+         "id":"09ec25f4-8fd4-11e8-9d5d-acde48001122",
+         "name":"Failed, Succeeded, Running, Stopped runs - Jul 24, 2018 22:29pm to Jul 24, 2018 23:29pm",
+         "created":1532500149,
+         "expiry":1532672965,
+         "status":"COMPLETED"
+      }
+   ]
+ }
+
+Sharing Reports
+---------------
+
+On Authentication enabled clusters, reports are isolated by logged-in users who generated the reports,
+If the user wishes to share a generated report, they can get an encrypted
```

Comment:
**If you wish** ..., **you can generate** an ... share-id **that can be used by others to access the report.**